services/queue/src/ec2regionresolver.js

```diff
@@ -9,19 +9,59 @@ const AWS_IP_RANGES_URL = 'https://ip-ranges.amazonaws.com/ip-ranges.json';
 /** Resolve EC2 region from request using AWS ip-ranges */
 class EC2RegionResolver {
   /** Construct EC2RegionResolver given a list of regions we care about */
-  constructor(regions) {
+  constructor(regions, monitor) {
     assert(regions instanceof Array, 'regions must be an array');
     this.regions = regions;
+    this.monitor = monitor;
     this.ipRanges = [];
   }
 
-  async loadIpRanges() {
-    // Get IP ranges from AWS with really stupid retry logic
-    let {body} = await request.get(AWS_IP_RANGES_URL).catch(() => {
-      return request.get(AWS_IP_RANGES_URL).catch(() => {
-        return request.get(AWS_IP_RANGES_URL);
-      });
+  /**
+   * Start running the resolver.  This will return immediately while loading the
+   * list of IP ranges.  Until that load is complete, the resolver will return
+   * null (meaning not an EC2 region) for all IPs.
```

Comment:
The implication being that nothing will use cloudmirror while this is happening? That seems like a great default but let's add that note to this comment.

Reply:
We never use cloudmirror (it's turned off).  But when the region data is in place we do hand out an S3 URL for requests from within the same region, and a CDN URL for everything else.